### ISSUE TITLE
migration to 2.5.1 and routers configuration

### ISSUE THREAD

**Issue by UriShohet** [akka/akka#22958]
Hi all, I'm trying to upgrade from akka 2.4.11 to 2.5.1 and I'm seeing something weird. I have cluster made of 1 master and 2 slave machines, but I also run the app from my laptop - a 1-node cluster. I I've defined a router in application.conf like this:
```
akka {
  actor.deployment {
    /restoreClient/workerRouter = {
      router = cluster-metrics-adaptive-pool
      metrics-selector = mix
      cluster {
        enabled = on
        use-role = "role1"
        allow-local-routees = on    # 'on' on localhost and 'off' when running on 3 machines.
        max-nr-of-instances-per-node = 10
      }
    }
  }
}
```
and I'm creating the router like this:
```scala
// from constructor of the actor named "restoreClient"
val router = context.actorOf(FromConfig.props(Props[RestoreSupervisorActor]), "workerRouter")
```
What's weird to me is that with akka 2.5.1 I must set ```akka.actor.deployment./restoreClient/workerRouter.cluster.enabled = off``` when running on 1-node cluster, otherwise the RestoreSupervisorActor actors are not created. When I send the GetRoutees message to the router, I get a list of 10 routees, but the messages I send to the router never get to the actors.
As soon as I go back to 2.4.11 or 2.4.18 - everything works fine again.
Was anything changed in 2.5.1 in this regard?
Thanks in advance,
Uri

**Comment by patriknw** [comment]
Thanks for reporting. Also good that you tested on 2.4.18. Sounds like a bug/regression from your description.

**Comment by UriShohet** [comment]
ok after even further investigation, I'd like to take it back and close the issue - the problem is not Akka but Kamon. 

I've checked again with 2.4.18 and it didn't work either until I downgraded Kamon back to 0.6.0

I've upgraded to kamon 0.6.6 while upgrading to akka 2.4.18 (and later to 2.5.1) and started to see issues. Then I went back to my previous branch with akka 2.4.11 and kamon 0.6.0 and there everything was OK, so I've opened this issue.

Anyway - I'm closing the issue now. I can't upgrade to 2.5.1 until kamon fix the [https://github.com/kamon-io/Kamon/issues/459](https://github.com/kamon-io/Kamon/issues/459)

Sorry about all the fuss
